packages/format-library/src/link/inline.js

```diff
@@ -113,71 +113,64 @@ function InlineLinkUI( {
-					type: 'core/link',
-				} );
-
-				// Split the value at the start of the active link format.
-				// Passing "start" as the 3rd parameter is required to ensure
-				// the second half of the split value is split at the format's
-				// start boundary and avoids relying on the value's "end" property
-				// which may not correspond correctly.
-				const [ valBefore, valAfter ] = split(
-					value,
-					boundary.start,
-					boundary.start
-				);
-
-				// Update the original (full) RichTextValue replacing the
-				// target text with the *new* RichTextValue containing:
-				// 1. The new text content.
-				// 2. The new link format.
-				// As "replace" will operate on the first match only, it is
-				// run only against the second half of the value which was
-				// split at the active format's boundary. This avoids a bug
-				// with incorrectly targetted replacements.
-				// See: https://github.com/WordPress/gutenberg/issues/41771.
-				// Note original formats will be lost when applying this change.
-				// That is expected behaviour.
-				// See: https://github.com/WordPress/gutenberg/pull/33849#issuecomment-936134179.
-				const newValAfter = replace( valAfter, richTextText, newValue );
-
-				newValue = concat( valBefore, newValAfter );
-			}
-
-			onChange( newValue );
+			// Scenario: Editing an existing link.
+
+			// Create new RichText value for the new text in order that we
+			// can apply formats to it.
+			newValue = create( { text: newText } );
+			// Apply the new Link format to this new text value.
+			newValue = applyFormat( newValue, linkFormat, 0, newText.length );
+
+			// Get the boundaries of the active link format.
+			const boundary = getFormatBoundary( value, {
+				type: 'core/link',
+			} );
+
+			// Split the value at the start of the active link format.
+			// Passing "start" as the 3rd parameter is required to ensure
+			// the second half of the split value is split at the format's
+			// start boundary and avoids relying on the value's "end" property
+			// which may not correspond correctly.
+			const [ valBefore, valAfter ] = split(
+				value,
+				boundary.start,
+				boundary.start
+			);
+
+			// Update the original (full) RichTextValue replacing the
+			// target text with the *new* RichTextValue containing:
+			// 1. The new text content.
+			// 2. The new link format.
+			// As "replace" will operate on the first match only, it is
+			// run only against the second half of the value which was
+			// split at the active format's boundary. This avoids a bug
+			// with incorrectly targetted replacements.
+			// See: https://github.com/WordPress/gutenberg/issues/41771.
+			// Note original formats will be lost when applying this change.
+			// That is expected behaviour.
+			// See: https://github.com/WordPress/gutenberg/pull/33849#issuecomment-936134179.
+			const newValAfter = replace( valAfter, richTextText, newValue );
+
+			newValue = concat( valBefore, newValAfter );
```

Comment:
I did not change any of these lines. The linter was complaining about the if / else / if / else, so I switched the format.